edgeless_api/proto/messages.proto

```diff
@@ -63,6 +63,54 @@ message SpawnFunctionResponse {
     optional InstanceId instance_id = 2;
 }
 
+// Possible message types of FunctionInstance::UpdateNode().
+enum UpdateNodeRequestType {
```

Comment:
I thought about that a lot.

Actually, I started with two different methods, but then I realised that we might also have to "update" a node, i.e., keeping the same node_id but changing the agent/invocation URLs.

It's a corner case but it can be supported easily with a single `UpdateNode()` message, while it would need a third method if we had something like `RegisterNode()` and `DeregisterNode()`.

Anyway, I don't have a strong opinion: if we ever feel like splitting the method into N others, it's fine (and it takes very little time).